### PR TITLE
[poap] POAP Fixes

### PR DESCRIPTION
**Pull request by gawainb**
Fixes #977 and #867 .

Changes proposed in this pull request:

- 1:1 strategy should also be capable of only counting specific tokens per their event id
- will use the erc721 strategy if no event ids passed
- with-weight block/snapshot param may not have been returning expected results

### CODE REVIEW

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/strategies/poap-with-weight/index.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export async function strategy(
     (token) => token.id
   );
   if (snapshot !== 'latest') {
-    getTokenSupply.tokens.__args.block = { number: snapshot };
+    getTokenSupply.tokens.__args['block'] = { number: snapshot };
```
Comment (ChaituVR) [inline src/strategies/poap-with-weight/index.ts:49]:
```suggestion
    getTokenSupply.tokens.__args.block = { number: snapshot };
```

File: src/strategies/poap-with-weight/index.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { subgraphRequest } from '../../utils';
 import examplesFile from './examples.json';
 
 export const author = 'gawainb';
-export const version = '1.1.0';
+export const version = '1.2.0';
```
Comment (ChaituVR) [inline src/strategies/poap-with-weight/index.ts:6]:
```suggestion
export const version = '1.1.0';
```

File: src/strategies/poap-with-weight/index.ts [hunk 3]
```diff
@@ -46,7 +46,7 @@ export async function strategy(
     (token) => token.id
   );
   if (snapshot !== 'latest') {
-    getTokenSupply.tokens.__args.block = { number: snapshot };
```
[the remark this answers is not in the document]
Comment (gawainb) [reply src/strategies/poap-with-weight/index.ts:49]: @greenealexander looks like this is ok the way it is 🙂

[the remark this answers is not in the document]
Comment (greenealexander) [reply src/strategies/poap-with-weight/index.ts:49]: @gawainb yeah, this can be reverted

**Review by ChaituVR — APPROVED** [review]

**Pull request merged**